packages/compiler/src/compiler/readMetadata.ts

```diff
@@ -64,20 +65,25 @@ export class ReadMetadata {
   constructor({
     document,
     referenceFn,
+    withParameters,
   }: {
     document: Document
     referenceFn?: ReferencePromptFn
-    configSchema?: object
+    withParameters?: string[]
   }) {
```

Comment:
`ReadMetadata` now accepts a `withParameters` optional prop.
When it is not defined, all variables that are being used in the code but have not been defined will be included in the returned metadata as `parameters`.
When defined, only variables included in this array can be used in the code. If a variable is used but was not previously defined and is not included in the array, an error is added to the returned metadata.